deepola/wake/src/polars_operations/reader/csvreader.rs

```diff
@@ -149,13 +149,19 @@ impl StreamProcessor<DataFrame> for CSVReader {
                         // This must be a length-one Polars series containing
                         // file names in its rows
                         let rows = series.utf8().unwrap();
+                        let num_files = rows.len();
 
                         // each file name produces multiple Series (each is a column)
-                        for filename in rows {
+                        for (index,filename) in rows.into_iter().enumerate() {
                             let output_df = self.dataframe_from_filename(filename.unwrap());
 
                             // Update record count
                             currect_total_records += output_df.height() as f64;
+                            if index == num_files - 1 {
+                                // lineitem's cardinality estimate of 6_000_000 * SF is approximate.
+                                // For some scales, this value is smaller than 6_000_000 * SF.
+                                currect_total_records = expected_total_records;
```

Comment:
More "correct" way would be to update the expected_total_records and metadata to the current + last df size. Pls ignore if needs a lot of changes